planner/core/optimizer.go

```diff
@@ -743,16 +744,29 @@ func calculateTiFlashStreamCountUsingMinLogicalCores(ctx context.Context, sctx s
 		return false, 0
 	}
 	var initialMaxCores uint64 = 10000
-	var minLogicalCores uint64 = initialMaxCores // set to a large enough value here
+	// set to a large enough value here
+	var minLogicalCores uint64 = initialMaxCores
+	var minPhysicalCores uint64 = initialMaxCores
+
 	for _, row := range rows {
 		if row[4].GetString() == "cpu-logical-cores" {
 			logicalCpus, err := strconv.Atoi(row[5].GetString())
-			if err == nil && logicalCpus > 0 && uint64(logicalCpus) < minLogicalCores {
-				minLogicalCores = uint64(logicalCpus)
+			if err == nil && logicalCpus > 0 {
+				minLogicalCores = mathutil.Min(minLogicalCores, uint64(logicalCpus))
+			}
+		}
+		if row[4].GetString() == "cpu-physical-cores" {
+			physicalCores, err := strconv.Atoi(row[5].GetString())
+			if err == nil && physicalCores > 0 {
+				minPhysicalCores = mathutil.Min(minPhysicalCores, uint64(physicalCores))
 			}
 		}
 	}
 	// No need to check len(serersInfo) == serverCount here, since missing some servers' info won't affect the correctness
+	if minPhysicalCores > 1 && minPhysicalCores != initialMaxCores {
```

Comment:
Not sure if physicalCores are total physical cores or total physical cores used by tiflash server

Reply:
OK. 
Use `runtime.GOARCH` to check platforms.